### PR TITLE
Resolving multiple problems with Linked Posts UI for Add/Remove

### PR DESCRIPTION
**Pull request by bordoni**
There were a lot of unnecessary code around the Linked Post duplication and removal.

@borkweb if you can please test this if you get the chance.

📹  https://i.bordoni.me/ZKMysqsY

---

Video With the Demo of the whole behavior working requested by @borkweb after some extra changes.

📹  https://i.bordoni.me/xBMb2qQs

---

### Community Events

📹 https://i.bordoni.me/L1dT9G5v

### CODE REVIEW

File: src/resources/js/event-editor.js [hunk 1]
```diff
@@ -12,101 +12,10 @@ var tribe_events_event_editor = tribe_events_event_editor || {};
...
-	/**
-	 * Controls logic for the Organizer delete button to display.
-	 *
-	 * If more than one organizer exists, display the delete button.
-	 * If only one organizer exists, hide the delete button.
-	 *
-	 * @since 6.0.1
-	 * @return void
-	 */
-	obj.linkedPost.deleteButtonDisplayLogic = function () {
```
Comment (borkweb) [inline src/resources/js/event-editor.js:36]: This is all of the logic for hiding and displaying the trash and add buttons, which is still needed. 

When there are no venues or organizers on the post, the add and trash buttons should not be visible. 

When there is a filled in venue or organizer, the button should be visible.

Venues should only have and add venue button and trash button when multi venues are enabled.

File: src/resources/js/events-admin.js [hunk 2]
```diff
@@ -619,10 +629,18 @@ jQuery( function( $ ) {
...
+		if ( 'tribe_venue' !== post_type ) {
+			return;
+		}
+
+		new_section.find( '#EventCountry' ).trigger( 'change' );
+	} );
+
 	//show state/province input based on first option in countries list, or based on user input of country
-	$( 'body' ).on( 'change', '#EventCountry', function () {
+	$( document ).on( 'change', '[id="EventCountry"]', function () {
```
Comment (borkweb) [inline src/resources/js/events-admin.js:641]: Searching by attribute is less efficient than searching by ID

**Review by borkweb — CHANGES_REQUESTED** [review]

File: src/resources/js/events-admin.js (hunk 2 above)
Comment (bordoni) [reply src/resources/js/events-admin.js:641]: Yeah but it breaks apparently, cause it only applies to the first one in the page. Cause IDs are supposed to be unique one per page.

File: src/resources/js/event-editor.js (hunk 1 above)
Comment (bordoni) [reply src/resources/js/event-editor.js:36]: Resolved these with the latest set of changes..

File: src/resources/js/events-admin.js [hunk 3]
```diff
@@ -364,17 +400,27 @@ jQuery( function( $ ) {
...
 			section.find( '.move-linked-post-group' ).hide();
 		}
+
+		section.find( 'select.linked-post-dropdown' ).trigger( 'change' );
 	};
 
 	var toggle_linked_post_fields = function( event ) {
-
 		const $select = $( this );
+		console.log( $select );
```
Comment (borkweb) [inline src/resources/js/events-admin.js:417]: Remove this guy!

File: src/resources/js/events-admin.js [hunk 4]
```diff
@@ -389,24 +435,20 @@ jQuery( function( $ ) {
...
-		) {
+		if ( ! existingPost && value ) {
 			// Apply the New Given Title to the Correct Field
 			$group.find( '.linked-post-name' ).val( value ).parents( '.linked-post' ).eq( 0 ).attr( 'data-hidden', true );
 
-			$select.val( '-1' );
+			$select.val( '' );
 
-			// Display the Fields
+			// Display thwe Fields
```
Comment (borkweb) [inline src/resources/js/events-admin.js:446]:
```suggestion
			// Display the Fields
```

**Review by borkweb — APPROVED** [review]
Just those two small things!

**Review by borkweb — APPROVED** [review]

**Pull request merged**